hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -555,6 +556,10 @@ SqlResult execute(SelectPlan plan, QueryId queryId, List<Object> arguments, long
                     }
 
                     partitionKeyComponents[i] = perMapCandidate.get(attribute).eval(null, evalContext);
+                    // handling case of __key = ? where the parameter is an PartitionAware OBJECT.
```

Comment:
in theory `AttributePartitionStragety` could use attribute that is `PartitionAware`. This is unlikely but valid. This logic should handle such case.

Reply:
We deliberately ignore PartitionAware altogether in the AttributeStrategy iirc. Don't quite remember why though. Either way I think it would be a bit of an extreme case where users would need to use PartitionAware fields within an Attribute-based key.

Reply:
I am thinking about the following case:
```java
class SomeVal extend PartitionAware {}
class Key { SomeVal keyVal; Integer whatever; Long notKey}
```
`IMap<Key, Object>` with `AttributePartitionStrategy` with `keyVal`. I am not saying that this makes big sense but we should calculate correct partition id in such case.

Reply:
this fails I am not sure how to fix that. It seems that for IMap put/get `AbstractSerializationService.calculatePartitionHash` gets attribute partitioning strategy, but for `getPartitionService().getPartition(...)` it test null/default strategy

Reply:
you can check the test in https://github.com/hazelcast/hazelcast/pull/25105/commits/8939e033dc856620b3ab170570add5320dfe00f2

Reply:
Fixed that in https://github.com/hazelcast/hazelcast/pull/25105/commits/f4f73af08f8f7417ba4675e3ebe4b65f9d7000c8, but I don't like this. WDYT?